test/app/test/core.clj

```diff
@@ -148,6 +153,56 @@
   (let [cl (parseType "ClassWithCallToFinalizeWithParams")]
     (is (false? (calls-finalizers? cl)))))
 
+(deftest testClassThatOverridesToString
```

Comment:
I think these tests cover all the cases, maybe you could just add a comment introducing the bunch of related tests (like the one for usageError at line 18